### ISSUE TITLE
Better handling of NULL values in Laminas\I18n\Filter\NumberParse

### ISSUE THREAD

**Issue by fabiang** [laminas/laminas-i18n#105]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | unsure

#### Summary

Hello,

since PHP 8.1 it's not possible to pass `null` to `NumberFormatter::parse()`:

```
Deprecated: NumberFormatter::parse(): Passing null to parameter #1 ($string) of type string is deprecated
```

see https://onlinephp.io/c/b9443.

In one of my forms I want an empty field to be parsed to `null` and otherwise to be parsed by `Laminas\I18n\Filter\NumberParse` filter. The filter chain looks like the following:

```php

use Laminas\I18n\Validator as I18nValidator;
use Laminas\I18n\Filter as I18nFilter;

return [
    [
        'name' => Filter\StringTrim::class,
    ],
    [
        'name'    => I18nFilter\NumberParse::class,
        'options' => [
            'locale' => $locale,
        ],
    ],
    [
        'name' => Filter\ToNull::class,
    ],
];
```

Problem here is that I bind an object to the form where the field value is `null` and the filter triggers the deprecation notice when I do `isValid()` on the form.

Therefor I propose to add a new option to the filter to allow null values to be passed to `filter()`.

Full test script:

```php
<?php

use Laminas\Form\Form;
use Laminas\Form\Element;
use Laminas\InputFilter\InputFilterProviderInterface;
use Laminas\I18n\Filter as I18nFilter;
use Laminas\Filter;

require 'vendor/autoload.php';

error_reporting(-1);
ini_set('display_errors', '1');

$bindObj = new class() {
    public function getArrayCopy()
    {
        return [
            'test' => null,
        ];
    }
};

$myform = new class() extends Form implements InputFilterProviderInterface{
    public function init()
    {
        $this->add([
            'type' => Element\Text::class,
            'name' => 'test'
        ]);
    }

    public function getInputFilterSpecification()
    {
        return [
            'test' => [
                'required' => false,
                'filters'  => [
                    [
                        'name' => Filter\StringTrim::class,
                    ],
                    [
                        'name'    => I18nFilter\NumberParse::class,
                        'options' => [
                            'locale' => 'en_US',
                        ],
                    ],
                    [
                        'name' => Filter\ToNull::class,
                    ],
                ]
            ],
        ];
    }
};

$myform->bind($bindObj);

$myform->setData([]);
var_dump($myform->isValid());

$myform->setData(['test' => '1,000.25']);
var_dump($myform->isValid()); // ok

```

Result:
```
Deprecated: NumberFormatter::parse(): Passing null to parameter #1 ($string) of type string is deprecated in vendor/laminas/laminas-i18n/src/Filter/NumberParse.php on line 158

Call Stack:
    0.0003     361216   1. {main}() test.php:0
    0.0281     596600   2. Laminas\Form\Form->isValid() test.php:59
    0.0461     815360   3. Laminas\Form\Form->bindValues($values = ???, $validationGroup = ???) vendor/laminas/laminas-form/src/Form.php:478
    0.0461     815360   4. Laminas\InputFilter\BaseInputFilter->getValues() vendor/laminas/laminas-form/src/Form.php:338
    0.0461     815576   5. Laminas\InputFilter\Input->getValue() vendor/laminas/laminas-inputfilter/src/BaseInputFilter.php:423
    0.0461     815576   6. Laminas\Filter\FilterChain->filter($value = NULL) vendor/laminas/laminas-inputfilter/src/Input.php:278
    0.0462     817888   7. Laminas\I18n\Filter\NumberParse->filter($value = NULL) vendor/laminas/laminas-filter/src/FilterChain.php:245
    0.0471     818880   8. NumberFormatter->parse($string = NULL, $type = 3) vendor/laminas/laminas-i18n/src/Filter/NumberParse.php:158

test.php:59:
bool(true)
test.php:62:
bool(true)

```

**Comment by froschdesign** [comment]
@fabiang 
Thanks for reporting the problem. If you can create a pull request with a unit test with/without a fix, that would be great. I will be happy to review it.
Thanks for your help! 👍🏻

**Comment by fabiang** [comment]
@froschdesign should we just cast the value to string? `NumberFormatter::parse()` then just returns false.

**Comment by froschdesign** [comment]
@fabiang 
The following is used in the filter `NumberFormat`:

https://github.com/laminas/laminas-i18n/blob/bae6888f2ab8e59629f74565856304d8cf2fec2c/src/Filter/NumberFormat.php#L21-L29

**Comment by gsteel** [comment]
Closed via #106
